File: Miscelaneous Questions/Algorithms in C#/isLeapYear.cs
using System;

namespace LeapYear
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Please enter a year:");
            int year = int.Parse(Console.ReadLine());
            bool isLeapYear = false;

            if (year % 4 == 0)
            {
                if (year % 100 == 0)
                {
                    if (year % 400 == 0)
                    {
                        isLeapYear = true;
                    }
                }
                else
                {
                    isLeapYear = true;
                }
            }

            if (isLeapYear)
            {
                Console.WriteLine($"{year} is a leap year.");
            }
            else
            {
                Console.WriteLine($"{year} is not a leap year.");
            }
        }
    }
}

File: Miscellaneous-Questions/Algorithms in C/isLeapYear.c
#include <stdio.h>

int IsLeapYear(int year);

int main() {
    int year;
	
    printf("Please enter the year: ");
    scanf("%d", &year);

    if (IsLeapYear(year)) {
        printf("%d is a leap year\n", year);
    } else {
        printf("%d is not a leap year\n", year);
    }

    return 0;
}

int IsLeapYear(int year) {
    if (year % 4 == 0) {
        if (year % 100 == 0) {
            if (year % 400 == 0)
                return 1;
            else
                return 0;
        } else {
            return 1;
        }
    } else {
        return 0;
    }
}
